homeassistant/components/sensor/meteo_france.py

```diff
@@ -0,0 +1,148 @@
+"""
+Support for Meteo France raining forecast.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.meteofrance/
+"""
+
+import logging
+import datetime
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+
+from homeassistant.const import (
+    STATE_UNKNOWN, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+_RESOURCE = 'http://www.meteofrance.com/mf3-rpc-portlet/rest/pluie/{}/'
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by Meteo France"
+CONF_LOCATION_ID = 'location_id'
+CONF_NAME = 'name'
+
+STATE_ATTR_FORECAST = 'Forecast'
+STATE_ATTR_FORECAST_INTERVAL = 'Interval_'
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(minutes=5)
+DEFAULT_TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_LOCATION_ID): cv.string,
+    vol.Optional(CONF_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Météo-France sensor."""
+    location_id = config.get(CONF_LOCATION_ID)
+    if len(location_id) == 5:  # convert insee code to needed meteofrance code
+        location_id = str(location_id)+'0'
+    location_name = config.get(CONF_NAME)
+    meteofrance_data = MeteoFranceCurrentData(hass, location_id)
+
+    try:
+        meteofrance_data.update()
+    except ValueError as err:
+        _LOGGER.error("Received error from Meteo France: %s", err)
+        return False
+
+    add_devices([MeteoFranceSensor(meteofrance_data, location_name)])
+
+
+class MeteoFranceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, meteofrance_data, location_name):
+        """Initialize the sensor."""
+        self.current_data = meteofrance_data
+        self.location_name = location_name
+        self._unit = "Min"
+        self._time_to_rain = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self.location_name is None:
+            return "1hr rain forecast"
+        return "{} 1hr rain forecast".format(self.location_name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.current_data and self.current_data.rain_forecast:
+            self._time_to_rain = 0
+            for interval in self.current_data.intervals:
+                if interval["niveauPluie"] > 1:
+                    self._unit = "Min"
+                    return self._time_to_rain
+                self._time_to_rain += 5
+            self._unit = ""
+            return "No rain"
```

Comment:
I understand but what is displayed here is the time to the next expected rain.
If it's already raining or if it will start in the next 5 minutes, then the value returned is 0.
What value should be returned if there is no rain for the next 60 minutes then ? Certainly not zero, and returning an int like 60 or 100 doesn't make much sense either.